### ISSUE TITLE
Specification uses inconsistent versions of RDF 

### ISSUE THREAD

**Issue by sesuncedu** [w3c/microdata-rdf#16]
Parts of the specification are written in terms of RDF 1.0 (e.g RDF-CONCEPTS) ; other parts are written in terms of RDF 1.1. The main issues arise with reference to plain literals (rdf:PlainLiteral) rather than simple literals (rdf:langString or xsd:string).

**Comment by gkellogg** [comment]
Good catch! @iherman is this serious enough to abort the publication? IMO, it doesn't change the interpretation of anything in the spec appreciably, but it would be cleaner to reference RDF11-CONCEPTS.

**Comment by iherman** [comment]
> On 15 Dec 2014, at 01:45 , Gregg Kellogg notifications@github.com wrote:
> 
> Good catch! @iherman is this serious enough to abort the publication? IMO, it doesn't change the interpretation of anything in the spec appreciably, but it would be cleaner to reference RDF11-CONCEPTS.

Publication is only tomorrow... and I agree it does not change anything on the implementations.

I have
- changed the references to rdf11 concepts
- added a separate term definition for plain literals
- changed the term definition of literals

I have pushed all the changes both on /TR and github. Have a look to see if there is a mistake.

Ivan

> —
> Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by sesuncedu** [comment]
Change _plain_ to _simple_ and it's got the right rdf 1.1 meaning, and the
situation for OWL is no worse than before.  Simple literals require awkward
kludge for OWL; plainliterals in RDF 1.1 are wrapped.

I will not explain why I know so much I know about this problem :)

**Comment by iherman** [comment]
I would prefer to leave it as 'plain' literal in the current version at least. For many people in the RDF community the term 'plain literal' is known from the earlier version of RDF and, in practice, what we have right now is related to that notion as well as precise in terms of RDF 1.1.

In practical terms, the document should be published tomorrow, there isn't too much time to make any change. I would hate to delay publication now unless it is really necessary:-)

ivan

> On 15 Dec 2014, at 17:23 , Simon Spero notifications@github.com wrote:
> 
> Change _plain_ to _simple_ and it's got the right rdf 1.1 meaning, and the
> situation for OWL is no worse than before. Simple literals require awkward
> kludge for OWL; plainliterals in RDF 1.1 are wrapped.
> 
> I will not explain why I know so much I know about this problem :)
> On Dec 15, 2014 5:53 AM, "Ivan Herman" notifications@github.com wrote:
> 
> > > On 15 Dec 2014, at 01:45 , Gregg Kellogg notifications@github.com
> > > wrote:
> > > 
> > > Good catch! @iherman is this serious enough to abort the publication?
> > > IMO, it doesn't change the interpretation of anything in the spec
> > > appreciably, but it would be cleaner to reference RDF11-CONCEPTS.
> > 
> > Publication is only tomorrow... and I agree it does not change anything on
> > the implementations.
> > 
> > I have
> > - changed the references to rdf11 concepts
> > - added a separate term definition for plain literals
> > - changed the term definition of literals
> > 
> > I have pushed all the changes both on /TR and github. Have a look to see
> > if there is a mistake.
> > 
> > Ivan
> > 
> > > —
> > > Reply to this email directly or view it on GitHub.
> > 
> > ---
> > 
> > Ivan Herman, W3C
> > Digital Publishing Activity Lead
> > Home: http://www.w3.org/People/Ivan/
> > mobile: +31-641044153
> > ORCID ID: http://orcid.org/0000-0003-0782-2704
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/w3c/microdata-rdf/issues/16#issuecomment-66977840.
> > 
> > —
> > Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by sesuncedu** [comment]
The minimal diff would be adding a note in the "algorithm terms" section
stating that the term "plain literal" corresponds precisely to the term
"simple literal" in RDF 1.1.

Specs like OWL 2 that are stuck with plain literals got messed up by the
simple literal change (owlapi 4 does some hidden magic in the bridge with
Rio, and I had to write a special comparator to handle the round trip
ambiguities.

The term "plain literal" for "literal without a datatype" is obsolete, so
using it in a live spec,  when the intended meaning is "simple literal" is
confusing.

If you are fixing the ref to go to the current RDF Concepts, search and
replace of plain to simple is wise, and avoids confusing people who are
attuned to the difference.

[I won't mention SPARQL 1.1 simple literals which are _almost_ the same as
RDF 1.1, save for a reference to simple literals sorting before typed
literals (ANSTAUL:).]

(fixing RDF to deal with literals as n-tuples properly would have made the
problem easier, because the specs probably wouldn't have reached LC yet :-)

**Comment by sesuncedu** [comment]
In other words I would treat plain as a typo of simple, not requiring a
revote.

**Comment by iherman** [comment]
I see; you are right. I must admit I did not remember that the term simple literal is used in the RDF 1.1

I have made a change from plain literal to simple literal everywhere in the document. This is now up on the servers.

Gregg, can you make a sanity check while I am asleep?

Thanks

Ivan

> On 15 Dec 2014, at 18:35 , Simon Spero notifications@github.com wrote:
> 
> In other words I would treat plain as a typo of simple, not requiring a
> revote.
> —
> Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by gkellogg** [comment]
No, we can't simply change to _simple literal_, as in RDF11-CONCEPTS, that specifically leaves out language-tagged literals:

> Please note that concrete syntaxes may support simple literals consisting of only a lexical form without any datatype IRI or language tag. Simple literals are syntactic sugar for abstract syntax literals with the datatype IRI http://www.w3.org/2001/XMLSchema#string. Similarly, most concrete syntaxes represent language-tagged strings without the datatype IRI because it always equals http://www.w3.org/1999/02/22-rdf-syntax-ns#langString.

In some places we can use either simple, or language-tagged literals. I'll update the text accordingly.

**Comment by sesuncedu** [comment]
Oh, you are quite right. Misremembering the two special cases; simple
literals is just the shorthand for xsd string and don't exist in the
model.

The only document  to that mentions plain literals is rdf semantics,  in
passim, and in a normative reference to the rdf:PlainLiteral spec.

**Comment by iherman** [comment]
Sigh. Last minute changes are always dangerous. Anyway, I am ahead of you, so tomorrow morning my time I can update /TR. Do not forget to regenerate the diff file.

That being said... the current version defines simple literals without referring to concepts. Ie, the current _is_ consistent...

:-(

Ivan

---

Ivan Herman
Tel:+31 641044153
http://www.ivan-herman.net

(Written on mobile, sorry for brevity and misspellings...)

**Comment by sesuncedu** [comment]
I literally hate literals.

**Comment by gkellogg** [comment]
I updated all the documents, including Overview and Diff. I ended up removing our own language on simple literals and just reference concepts for both simple literals and language-tagged strings. I didn't see value in re-defining those concepts. Give it a glance, if you can.

**Comment by sesuncedu** [comment]
The typing of the default case for `<data>` may have changed, since specifying that a type was a "plain literal" implies that the value may be "text" (which could make "lang" tag  apply).  

Since a  simple literal is syntactic sugar for xsd:string, and since this seems to be a better fit, I would call this an editorial clarification.

**Comment by sesuncedu** [comment]
The descriptions of Tests 77 and and 80 explicitly refer to `string` and `xsd:string` respectively (however, there is no lang attribute in scope that could affect the result).  The expected results are serialized as  "simple literals", but stripping explicit xsd:string typing is  permitted under RDF 1.1 for most concrete syntaxes (and is generally the default).

**Comment by iherman** [comment]
Gregg,

thanks! Yes, your version is better that way.

For some reasons, the diff did not make it to the repo. I did generated it, and updated the repo accordingly. And I updated the version on /TR @W3C

This IS the last version. I am not sure when Jérémie will publish (I do not know whether he is still in the US or whether he came back to Europe for the holidays), so this is now _frozen_. As soon as the publication is done, I will also update the registry file.

I would propose if we have any new issues then, well, we should push them as a new issue. I will also look at the issues on github to close all issues that, in some way or other, have been taken care of. I would propose any new problem should be added as, well a completely new issue to keep this for later.

Thanks to all. And many thanks to Simon to make us realize that there was an inconsistency in the RDF reference; I feel stupid not to have thought of that earlier!

Ivan

> On 15 Dec 2014, at 22:24 , Gregg Kellogg notifications@github.com wrote:
> 
> I updated all the documents, including Overview and Diff. I ended up removing our own language on simple literals and just reference concepts for both simple literals and language-tagged strings. I didn't see value in re-defining those concepts. Give it a glance, if you can.
> 
> —
> Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704
